openslides/utils/autoupdate.py

```diff
@@ -72,78 +72,58 @@ def on_open(self, info):
-                wsgi_network_location = 'http://localhost:{}'.format(RUNNING_PORT)
-            else:
-                wsgi_network_location = 'http://{}:{}'.format(RUNNING_HOST, RUNNING_PORT)
-        url = ''.join((wsgi_network_location, object_url))
+            raise ValueError('Invalid container. A valid dispatch_uid is missing.')
 
-        # Send out internal HTTP request to get data from the REST api.
+        # Loop over all waiters
         for waiter in cls.waiters:
-            # Initiat new headers object.
-            headers = HTTPHeaders()
-
-            # Read waiter's former cookies and parse session cookie to new header object.
+            # Read waiter's former cookies and parse session cookie to get user instance.
             try:
                 session_cookie = waiter.connection_info.cookies[settings.SESSION_COOKIE_NAME]
             except KeyError:
-                # There is no session cookie
-                pass
+                # There is no session cookie so use anonymous user here.
+                user = AnonymousUser()
             else:
-                headers.add('Cookie', '%s=%s' % (settings.SESSION_COOKIE_NAME, session_cookie.value))
-
-            # Read waiter's language header.
-            try:
-                languages = waiter.connection_info.headers['Accept-Language']
-            except KeyError:
-                # There is no language header
-                pass
+                # Get session from session store and use it to retrieve the user.
+                engine = import_module(settings.SESSION_ENGINE)
+                session = engine.SessionStore(session_cookie.value)
+                fake_request = type('FakeRequest', (), {})()
+                fake_request.session = session
+                user = get_user(fake_request)
+
+            # Two cases: models instance was changed or deleted
+            if container.get('action') == 'changed':
+                data = access_permissions.get_restricted_data(container.get('full_data'), user)
+                if data is None:
+                    # There are no data for the user so he can't see the object. Skip him.
+                    break
+                output = {
+                    'status_code': 200,  # TODO: Refactor this. Use strings like 'change' or 'delete'.
+                    'collection': container['collection_string'],
+                    'id': container['rest_pk'],
+                    'data': data}
+            elif container.get('action') == 'deleted':
+                output = {
+                    'status_code': 404,  # TODO: Refactor this. Use strings like 'change' or 'delete'.
+                    'collection': container['collection_string'],
+                    'id': container['rest_pk']}
             else:
-                headers.parse_line('Accept-Language: ' + languages)
+                raise ValueError('Invalid container. A valid action is missing.')
```

Comment:
I tested this error: If it is raised the client gets a HTTP 500 but the server continues running even with tornado debug set to False. So I think this is ok. I expect this error only if a developer writes bad code so that wrong data are submitted to our autoupdate system.